### PR TITLE
Add protobuf comments for oracle modules

### PR DESCRIPTION
**Pull request by ntchjb**
This PR adds comments to Protobuf files to provide more information and help understanding data structure of the Protobuf messages.

## Implementation details

**No code, only comments.**

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: proto/oracle/v1/oracle.proto [hunk 1]
```diff
@@ -227,7 +270,11 @@ message Result {
 // ValidatorStatus maintains whether a validator is an active oracle provider.
 message ValidatorStatus {
   option (gogoproto.equal) = true;
+  // IsActive is a boolean indicating active status of validator.
+  // The validator will be deactivated when they are unable to send reports
+  // to fulfill oracle request.
```
Comment (thebevrishot) [inline proto/oracle/v1/oracle.proto:275]: before the request expired

File: proto/oracle/v1/oracle.proto [hunk 2]
```diff
@@ -261,18 +308,19 @@ message Params {
   uint64 inactive_penalty_duration = 8;
 }
 
-// PendingResolveList
+// PendingResolveList is a list of requests that are waiting to be resolved by validators
```
Comment (thebevrishot) [inline proto/oracle/v1/oracle.proto:311]: Requests would be resolved by any nodes and the results would be stored locally.

File: proto/oracle/v1/oracle.proto [hunk 3]
```diff
@@ -261,18 +308,19 @@ message Params {
...
 message PendingResolveList {
   option (gogoproto.equal) = true;
+  // RequestIDs is a list of request IDs that are waiting to be resolved by validators
```
Comment (thebevrishot) [inline proto/oracle/v1/oracle.proto:314]: please edit this one too

File: proto/oracle/v1/oracle.proto [hunk 4]
```diff
@@ -261,18 +308,19 @@ message Params {
...
   repeated int64 request_ids = 1;
 }
 
-// IBCSource
+// IBCSource is information of IBC protocol to allow communicating with other chain
 message IBCSource {
   option (gogoproto.equal) = true;
 
-  // SourceChannel
+  // SourceChannel is channel ID used for communicating with other chain
```
Comment (thebevrishot) [inline proto/oracle/v1/oracle.proto:322]: chains?

File: proto/oracle/v1/oracle.proto [hunk 5]
```diff
@@ -261,18 +308,19 @@ message Params {
...
   string source_channel = 1;
-  // SourcePort
+  // SourcePort is port ID used for communicating with other chain
```
Comment (thebevrishot) [inline proto/oracle/v1/oracle.proto:324]: ditto

File: proto/oracle/v1/query.proto [hunk 6]
```diff
@@ -120,31 +146,50 @@ message QueryParamsResponse {
...
 // QueryRequestSearchRequest is request type for the Query/RequestSearch RPC
 // method.
 message QueryRequestSearchRequest {
+  // OracleScriptID represents ID of an oracle script
   int64 oracle_script_id = 1;
+  // Calldata represents OBI-encoded data as argument params for the oracle script
   bytes calldata = 2;
+  // AskCount represents number of validators allowed for fulfilling the request
   int64 ask_count = 3;
+  // MinCount represents number of validators required to fulfill the request
```
Comment (thebevrishot) [inline proto/oracle/v1/query.proto:192]: could we use the same format as upper for consistency

File: proto/oracle/v1/query.proto [hunk 7]
```diff
@@ -157,16 +202,23 @@ message QueryRequestSearchResponse {
 // QueryRequestPriceRequest is request type for the Query/RequestPrice RPC
 // method.
 message QueryRequestPriceRequest {
+  // Symbol represents currency unit
```
Comment (thebevrishot) [inline proto/oracle/v1/query.proto:205]: it's not just `currency`

File: proto/oracle/v1/query.proto [hunk 8]
```diff
@@ -157,16 +202,23 @@ message QueryRequestSearchResponse {
...
   string symbol = 1;
+  // AskCount represents number of validators allowed for fulfilling the request
   int64 ask_count = 2;
+  // MinCount represents number of validators required to fulfill the request
```
Comment (thebevrishot) [inline proto/oracle/v1/query.proto:209]: same question as above

File: proto/oracle/v1/query.proto [hunk 9]
```diff
@@ -157,16 +202,23 @@ message QueryRequestSearchResponse {
...
 // QueryRequestPriceResponse is response type for the Query/RequestPrice RPC
 // method.
 message QueryRequestPriceResponse {
+  // OracleScriptID represents ID of an oracle script
   int64 oracle_script_id = 1;
+  // Calldata represents OBI-encoded data as argument params for the oracle script
   bytes calldata = 2;
+  // AskCount represents number of validators allowed for fulfilling the request
   int64 ask_count = 3;
+  // MinCount represents number of validators required to fulfill the request
```
Comment (thebevrishot) [inline proto/oracle/v1/query.proto:222]: same question as above

File: proto/oracle/v1/tx.proto [hunk 10]
```diff
@@ -38,26 +38,26 @@ service Msg {
...
 message MsgRequestData {
   option (gogoproto.equal) = true;
-  // OracleScriptID is the identifier of the oracle script to call.
+  // OracleScriptID is the identifier of the oracle script to be called.
   int64 oracle_script_id = 1 [
     (gogoproto.customname) = "OracleScriptID",
     (gogoproto.casttype) = "OracleScriptID"
   ];
-  // Calldata is the OBI encoded call parameters to the oracle script.
+  // Calldata is the OBI encoded call parameters for the oracle script.
```
Comment (thebevrishot) [inline proto/oracle/v1/tx.proto:53]: should it be `OBI-encoded` same as above?

File: proto/oracle/v1/oracle.proto [hunk 11]
```diff
@@ -24,62 +31,97 @@ message DataSource {
...
 message OracleScript {
   option (gogoproto.equal) = true;
+  // Owner represents an address of the account who own the oracle script
   string owner = 1;
+  // Name is oracle script name used for display
   string name = 2;
+  // Description is oracle script description used for display
   string description = 3;
+  // Filename represents string of file name used as reference for locating
+  // compiled oracle script WASM file stored in validator's node
```
[the remark this answers is not in the document]
Comment (ntchjb) [reply proto/oracle/v1/oracle.proto]: fixed to `bandchain nodes`

File: proto/oracle/v1/oracle.proto [hunk 12]
```diff
@@ -11,10 +11,17 @@ import "cosmos/base/v1beta1/coin.proto";
...
 message DataSource {
   option (gogoproto.equal) = true;
+  // Owner represents an address of the account who own the data source
   string owner = 1;
+  // Name is data source name used for display
   string name = 2;
+  // Description is data source description used for display
   string description = 3;
+  // Filename represents string of file name used as reference for locating
+  // data source file stored in validator's node
```
[the remark this answers is not in the document]
Comment (ntchjb) [reply proto/oracle/v1/oracle.proto]: fixed to `bandchain nodes`

File: proto/oracle/v1/query.proto (hunk 7 above)
Comment (ntchjb) [reply proto/oracle/v1/query.proto]: Fixed to `unit of data indicating what the data is`

File: proto/oracle/v1/query.proto (hunk 8 above)
Comment (ntchjb) [reply proto/oracle/v1/query.proto]: Could you elaborate? Not sure if it is formatting or `validators`

Comment (thebevrishot) [reply proto/oracle/v1/query.proto]: I mean formatting krub. I think it should be `for fulfilling` to be consistent with the above comment.

**Review by thebevrishot — APPROVED** [review]
LGTM

**Review by Benzbeeb — APPROVED** [review]

**Pull request merged**